preload/subresource-integrity.html

```diff
@@ -309,6 +309,39 @@
         {integrity: "sha256-deaddeadbeefYHFvsYdWumweeFAw0hJDTFt9seErghA="}
       )
 
+      SRIPreloadTest(
+        true,
+        true,
+        `Same-origin ${destination} with matching digest does not reuse preload without digest.`,
+        2,
+        destination,
+        same_origin_prefix + destination + ext + `?${token()}`,
+        {},
+        {integrity: sha256}
+      )
+
+      SRIPreloadTest(
```

Comment:
This sub-test can fail, according to https://github.com/whatwg/html/pull/7738:

> The user-agent has determined that entry's integrity metadata's algorithm is more collision-resistant than integrityMetadata's algorithm

(and actually fails on Firefox; also failing on Safari, but because some other tests are failing on Safari I suspect there's an underlying bug there).

So it's better to mention in the description/comment etc. about the acceptable failure.

Reply:
Will do for firefox.
I opened a WebKit bug to cover all the failures, WebKit currently doesn't do integrity matching for preloads which can lead to invalid resources in certain situations.